sdk/python/tests/conftest.py

```diff
@@ -138,20 +141,29 @@ def simple_dataset_2() -> pd.DataFrame:
     params=FULL_REPO_CONFIGS, scope="session", ids=[str(c) for c in FULL_REPO_CONFIGS]
 )
 def environment(request):
-    with construct_test_environment(request.param) as e:
-        yield e
+    e = construct_test_environment(request.param)
+
+    def cleanup():
+        e.feature_store.teardown()
+
+    request.addfinalizer(cleanup)
```

Comment:
but isn't this code equivalent to the previous version with context managers?
I believe pytest under the hook does the same `addfinalizer`, https://github.com/pytest-dev/pytest/blob/main/src/_pytest/fixtures.py#L921